packages/contracts/contracts/ProxyUpdater.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+/**
+ * @title ProxyUpdater
+ * @notice The ProxyUpdater contains the logic that sets the bytecode and storage of the proxy
+ *         when an action is executed in the ChugSplashManager. When an action is executed, the
+ *         ChugSplashManager *temporarily* sets the proxy's implementation to be this contract
+ *         so that the proxy can delegatecall into it.
+ */
+contract ProxyUpdater {
+    /**
+     * @notice "Magic" prefix. When prepended to some arbitrary bytecode and used to create a
+     *         contract, the appended bytecode will be deployed as given.
+     */
+    bytes13 internal constant DEPLOY_CODE_PREFIX = 0x600D380380600D6000396000f3;
+
+    /**
+     * @notice Sets the code for the proxy's implementation. Since this contract *is* the proxy's
+     *         implementation at the time of this call, it will be overwritten by the new
+     *         implementation. This is fine, since the ProxyUpdater is only meant to be the
+     *         implementation contract temporarily. Also note that this scheme is a bit different
+     *         from the standard proxy scheme where one would typically deploy the code separately
+     *         and then set the implementation address. We're doing it this way because it gives
+     *         us a lot more freedom on the client side.
+     *
+     * @param _implementationKey The key of the proxy's implementation address. For EIP-1967
+     *                           proxies, this value is:
+     *                           `bytes32(uint256(keccak256('eip1967.proxy.implementation')) - 1)`.
+     * @param _code              New contract code to deploy.
+     */
+    function setCode(bytes32 _implementationKey, bytes memory _code) external {
```

Comment:
This contract may need to be made non-reentrant or possible `owned` if we add support for constructors where people can run arbitrary code. I think this is fine right now since `DEPLOY_CODE_PREFIX` guarantees that no sub-calls will be executed, but a constructor could in theory call another contract which in turn calls back to `setCode` or `setStorage`.

Reply:
We should at least have a TODO here.